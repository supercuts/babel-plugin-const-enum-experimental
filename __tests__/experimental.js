import { transformAsync } from '@babel/core';
import plugin from '../src';

const options = {
  plugins: [[plugin, { transform: 'experimental' }]],
};

it('works unstably', async () => {
  const input = `const enum Direction { Left, Right, Down, Up }
console.log(Direction.Left)`;

  const { code: output } = await transformAsync(input, options);
  expect(output).toMatchSnapshot();

});

it('works unstably starting with initializer', async () => {
  const input = `const enum Direction { Left = 1, Right, Down, Up }
console.log(Direction.Left)`;

  const { code: output } = await transformAsync(input, options);
  expect(output).toMatchSnapshot();

});

it('is empty if not in scope', async () => {
  const input = `{const enum Direction { Left, Right, Down, Up }}
console.log(Direction.Left)`;
  try {
    const { code: output } = await transformAsync(input, options);
    expect(1 === 2);
  } catch(e) {
    expect(1 === 1);
  }
});
