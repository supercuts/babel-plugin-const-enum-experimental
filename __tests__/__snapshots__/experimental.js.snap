// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works unstably 1`] = `"console.log(0);"`;

exports[`works unstably starting with initializer 1`] = `"console.log(0);"`;
